src/services/axios-wrapper.ts

```diff
@@ -5,7 +5,7 @@ import { notLoggedInException } from '../store/user/actions';
 const axiosInstance = axios.create();
 
 axiosInstance.interceptors.response.use((res: any) => {
-  if(res.error?.message === 'must be logged in') {
+  if(res?.error?.message === 'must be logged in') {
```

Comment:
please squash all your commits into one commit please - each commit should strive to be complete. in "complete" I mean - the code got reviewed and approved, the code is working as expected, tests were added if applicable. The reason to be strict on complete commits is that it is helpful to use the history (blame, annotations) during work, especially in debugging bugs and understanding older code. if the history is with a lot of garbage (a lot of meaningless, half-working, not-code-reviewed commits) you soon lose your faith in it and using history is less powerful.
There is a recommended book (I didnt read it) that talks about source control code as a crime scene to help you improve your project: https://pragprog.com/book/atcrime/your-code-as-a-crime-scene
:)

Reply:
I get what you mean and totally agree with you, i am now experiencing that on the project i am at now, git history is useless.
This is result of a mid knowledge of controlling the git, for example: i work on branch x and i created few changes on that branch. then i want to go to branch y and work on some stuff there, so what i do i commit on x with what i've done so far and returning to it later.

Maybe i should use stash instead in this cases. Promise to improve :)

Reply:
there is nothing wrong with half-baked commits - as long as they are not merged into master
so you can use the rebase mechanism to squash/fixup the commits.

Reply:
for rebasing and squashing commits this is what I do:
in terminal: `git rebase -i master`
vim is opened and lets say i have these commits:
```
pick 64750de not a valid commit message
pick c698bd4 cr fixes
pick 5f6c61d fix
```
turn this into:
```
r 64750de not a valid commit message
f c698bd4 cr fixes
f 5f6c61d fix
```
(f is fixup - meaning it will be merged with the commit from the top)
(r is for renaming commit message)
now hit `:wq` to close and save
now it will open like this:
```
not a valid commit message
```
and you change it to a valid commit message:
```
fix(widgets.actions.ts):publish complete
```
now hit `:wq` again and it is saved
you will need to force push whenever you rebase:
`git push --force`
please use commit message in this convention:
```
action(module or file of the essence of the change):a short description of a few words so we can understand what is this commit
```

Reply:
Thank you, i toke a day to learn deeper about the git rebase and the vim environment.
I Squashed it all to single commit, please accept review if it is as expected.